druid/src/widget/textbox.rs

```diff
@@ -606,12 +606,16 @@ impl<T: TextStorage + EditableText> Widget<T> for TextBox<T> {
 
             let padding_offset = Vec2::new(textbox_insets.x0, textbox_insets.y0);
 
-            let cursor = if data.is_empty() {
+            let mut cursor = if data.is_empty() {
                 cursor_line + padding_offset
             } else {
                 cursor_line + padding_offset - self.inner.offset()
             };
 
+            // Snap the cursor to the pixel grid so it stays sharp.
+            cursor.p0.x = cursor.p0.x.trunc() + 0.5;
+            cursor.p1.x = cursor.p0.x;
```

Comment:
- this should be correct for coregraphics and skia; i'm not sure if directwrite behaves the same way?
- might `round` make more sense than `trunc`?
- I'm actually also not sure if this will work at all scale factors. it should work at 1x and 2x? I'm less sure about other cases.


All that said: I'm not opposed to this at all, it's just exposing some of my general confusion about pixels and scale factor. I would really love to see a sort of 'unified' theory of this stuff; until we have that it's going to feel like our drawing code is an inconsistently applied series of heuristics. :/